api/overdrive.py

```diff
@@ -262,7 +278,12 @@ def get_fulfillment_link(self, patron, pin, overdrive_id, format_type):
             response = self.lock_in_format(
                 patron, pin, overdrive_id, format_type)
             if response.status_code not in (201, 200):
-                raise CannotFulfill("Could not lock in format %s" % format_type)
+                if response.status_code == 400:
+                    message = json.loads(response.content).get("message")
+                    if message == "The selected format may not be available for this title.":
+                        raise FormatNotAvailable("This book is not Available the requested format")
```

Comment:
Copyedit: "This book is not available in the format you requested."

I don't normally copyedit but this string will go out to patrons.